### ISSUE TITLE
Associate obs inlet with fp_height

### ISSUE THREAD

**Issue by joe-pitt** [openghg/openghg_inversions#220]
At the moment obs data for each site is selected by specifying an inlet list in the ini file. You then specify a separate fp_height list to get the right footprints. But this can't handle a case where the obs inlet changes halfway through a year, such as the TAC medusa in 2017. In this case, we want inlet=100m, fp_height=100m for January, then inlet=185m, fp_height=185m for the rest of the year. We can select a range of inlet heights for the obs using slice, but there isn't currently a way to associate the 100m fps with the 100m obs and the 185m fps with the 185m obs.

One possible solution would be to add duplicate entries of the same "site", but with different inlet and fp_height values. But the fp_all dictionary is not set up to handle this at the moment. So either this dictionary could be changed so that it distinguishes between different inlets at the same height, or some option to combine the data upfront could be added (e.g. changing fp_height in the ini file into a dictionary keyed by the obs inlet).

**Comment by joe-pitt** [comment]
Need to think about BSD too. For the MD you need to pick 108m fps before 2017-03-17 15;00, and 248m fps after that time. So maybe that would advocate for using a dictionary with the obs inlet as the keys, and either a list or a slice to select corresponding fp_heights for each obs inlet

**Comment by joe-pitt** [comment]
Adding a slightly separate case here because I think any solution will need to consider both. In both cases above obs stop at the lower height when they start at the higher height. But there are also cases (e.g. HUN and SSL CH4) where obs at the lower height continue after the higher height is added. At that point we want to switch to the higher height and ignore the lower height.

**Comment by brendan-m-murphy** [comment]
I think we'll need to make the changes in OpenGHG, around the "combine and elevate inlet" functionality in `get_obs_surface`. 

One potential challenge is if the obs. for different heights are not exactly aligned on time, then we can't select one over the other just by dropping duplicates. 

Is it always the case that if a new inlet is added, it will be higher up? That might simplify things because we could maybe use the start date for the top inlet height as the end date for other inlets.

**Comment by joe-pitt** [comment]
Hmmm unfortunately I don't think that will always be the case. Another option would be to make this more manual - i.e. the user can optionally specify a time period for each inlet height (rather than relying on the code to pick the top one automatically). That might be more robust actually.

**Comment by brendan-m-murphy** [comment]
Yea, it would be nice to be able to specify what data you want more precisely. One option for this would be to allow obs to be specified outside the current format of lists of sites, inlets, etc. Also I suppose we could add optional start/end dates for the obs.

I think actually we could probably combine the obs iteratively: combine the top two inlets by first dropping times from the lower inlet that are duplicates (within some tolerance), once those are combined, we could repeat the process to add lower inlets.

I suppose if we just do the case of two inlets, it might not be much work to extend it later to 3+ inlets.

**Comment by joe-pitt** [comment]
I think there are more than two heights at HUN, so we'd probably need to have the 3+ height option to start with. Would the manual method be easier or harder to implement that the automatic top height selection method? If it's no harder then perhaps we should start with that. Then we will have the flexibility to handle new cases as they arise (I'm already thinking about CGO which will probably be different again)

**Comment by hdelongueville** [comment]
Hasn't this issue been solved with the option `auto` in `fp_height`?

At least it works for TAC.

**Comment by joe-pitt** [comment]
I think the original case has been fixed but I don't think we have solved the issue at sites like HUN and SSL where a higher inlet gets added but the original (lower) inlet continues too

**Comment by brendan-m-murphy** [comment]
Hmm can we test that? I think the "auto" option should take the closest inlet to the obs height, so if extra footprint heights become available, they might not be used.

**Comment by brendan-m-murphy** [comment]
Oh so if the inlet is set with slice? It might be possible to prioritise higher inlet values.

**Comment by joe-pitt** [comment]
Yeah we had some discussion above about whether this should be done manually or automatically. I still think there might be an advantage to the manual approach - i.e. an option to specify a range of times for each inlet. Otherwise we might end up chopping between inlets in some cases
